### PR TITLE
added info about reactive and layered charms

### LINKED ISSUE #1274: Developer Getting started mentions reactive but offers no explanation on what it is
We have done a good job of putting the layer charming approach at the forefront but have not done anything aside from mentioning reactive on the  getting started guide for developers.

We mention it but offer no links to learn more or even a description of what it provides. 

for reference [Developer Getting Started](https://jujucharms.com/docs/stable/developer-getting-started#writing-your-charm)

>  The basic layer provides charm helpers python library and the reactive framework that makes layers possible

We should at bare minimum link to the reactive documentation on pythonhosted.

### PR DESCRIPTION
**Pull request by matthewhelmke**
Fixes #1274 
I also found a comma that should have been a full stop and fixed that in a separate file.
@juju/docs

### CODE REVIEW

File: src/en/developer-getting-started.md [hunk 1]
```diff
@@ -94,6 +94,26 @@ The reactive framework used with layered charms runs under Python 3, so you'll
 want to be aware of [compatibility](http://python-future.org/compatible_idioms.html)
 issues between Python 2 and 3.
 
+### Reactive and layered charms
+
+#### Reactive
+
+Another software paradigm is
+[reactive programming](https://en.wikipedia.org/wiki/Reactive_programming). Do
```
Comment (evilnick) [inline src/en/developer-getting-started.md]: can we put links at the bottom please!

File: src/en/developer-getting-started.md [hunk 2]
```diff
@@ -94,6 +94,26 @@ The reactive framework used with layered charms runs under Python 3, so you'll
...
+something when the state or conditions are correct. Juju offers the
```
Comment (evilnick) [inline src/en/developer-getting-started.md]: i would s/are correct/indicate

**Comment by evilnick** [comment]
I know that the document is littered with other links that aren't at the bottom, but i am fixing that :smiley:

**Comment by matthewhelmke** [comment]
The requested changes make sense and have been done.

**Pull request merged**